tools/firtool/firtool.cpp

```diff
@@ -407,8 +409,20 @@ static LogicalResult processBuffer(
     if (failed(parsePassPipeline(StringRef(lowFIRRTLPassPlugin), pm)))
       return failure();
 
-  // Lower if we are going to verilog or if lowering was specifically requested.
-  if (outputFormat != OutputIRFir) {
+  // Add passes specific to btor2 emission
+  if (outputFormat == OutputBTOR2) {
+    if (failed(firtool::populateLowFIRRTLToHW(pm, firtoolOptions)))
+      return failure();
+    if (!hwPassPlugin.empty())
+      if (failed(parsePassPipeline(StringRef(hwPassPlugin), pm)))
+        return failure();
```

Comment:
Can we hoist LowerFIRRTLToHW and hwPassPlugin part?

Reply:
yeah ofc